data.js

```diff
@@ -40,20 +40,63 @@
 */
 
 function rcxDict(loadNames) {
-	this.loadDictionary();
-	if (loadNames) this.loadNames();
-	this.loadDIF();
+
+	
 }
 
 rcxDict.prototype = {
 	config: {},
 
-	setConfig: function(c) {
+	init: function (loadNames) {
+		var started = +new Date();
+
+		var promises = [
+			this.loadDictionary(loadNames),
+			this.loadDIF()
+		];
+
+		return Promise.all(promises)
+			.then(function () {
+				var ended = +new Date();
+				console.log('rcxDict main then in ' + (ended - started));
+			}.bind(this));
+	},
+
+	setConfig: function (c) {
 		this.config = c;
 	},
 
 	//
 
+	fileReadAsync: function (url, asArray) {
+		return new Promise(function prom(resolve, reject) {
+			var req = new XMLHttpRequest();
+
+			req.onreadystatechange = function () {
+
+				if (req.readyState === 4) {
+
+					if (!asArray)
```

Comment:
it might be helpful for future editors if we use braces for ifs that precede elses.